Network/Curl/Opts.hs

```diff
@@ -265,14 +266,19 @@ type ReadFunction
  -> CInt       --  number of items
  -> Ptr ()     --  state argument (file pointer etc.)
  -> IO (Maybe CInt) --  how many bytes was copied into buffer; Nothing => abort.
- 
+
 type ReadFunctionPrim
   = Ptr CChar
  -> CInt
  -> CInt
  -> Ptr ()
  -> IO CInt
- 
+
+type OpenSocketFunction
+   = Ptr () -- ^ Pointer to clientp data set by OpenSocketData
+  -> CInt  -- ^ The curl socket type
+  -> Ptr () -- ^ The struct describing the curl_sockaddr struct
+  -> IO (Ptr ()) -- ^ The opened socket or CURL_SOCKET_BAD
```

Comment:
Wait, actually, why is this a `Ptr ()`, `curl_socket_t` is defined as `int` (https://github.com/curl/curl/blob/master/include/curl/curl.h#L137), because it returns a file descriptor, which is simply an int.

Reply:
Ah, yes, I was tripped up by the weird type signature in C and the fact that it worked, but I will change it.

Reply:
This has been changed!